### ISSUE TITLE
Only detect one event after of fitting

### ISSUE THREAD

**Issue by PARODBE** [autoreject/autoreject#223]
Hi!

Thank you for building such a useful library. I have a doubt, after of fitting:

![image](https://user-images.githubusercontent.com/62376157/128240070-6b3d9b7b-3101-4cb3-9b09-cbb803d72ef6.png)

I only obtain 1 event:

![image](https://user-images.githubusercontent.com/62376157/128240165-a4028dc7-a357-466f-9e75-09a90ed49556.png)

And I Know that we have much more events.

![image](https://user-images.githubusercontent.com/62376157/128240388-62e15715-72ee-4d7a-9211-dba8f32d84de.png)

Am I doing something bad? Could you help me please?

Thanks!

**Comment by jasmainak** [comment]
@PARODBE it says: Number of events = 762. Am I missing something? What events do you expect that are missing?

**Comment by PARODBE** [comment]
Sorry @jasmainak I have not been clear,  I want to soy annotations. Could you help me on extracting the annotations .. sleep stage w, Sleep Stage N1 etc...

Thanks

**Comment by PARODBE** [comment]
The epochs have a function for extract the annotations. Epochs.event_id...but If I do epochs_clean with this library .event, I only obtain one annotation (1)...

**Comment by jasmainak** [comment]
please share a code snippet that I can run on my computer so I understand your problem

**Comment by PARODBE** [comment]
Nice,

When the next code you obtain your data and your label:

from numpy import save
import mne
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.preprocessing import StandardScaler

data = mne.io.read_raw_edf('/media/usbdisk/data/ProyectoPSG/data/PSG1.edf', preload=True)
raw_data = data.get_data()
events, events_id = mne.events_from_annotations(data)
tmax = 30. - 1. / data.info['sfreq']  # tmax in included
epochs_train = mne.Epochs(raw=data, events=events,
                              event_id=events_id, tmin=0., tmax=tmax, baseline=None, event_repeated='merge',preload=True)

You can see the information in epochs train:

![image](https://user-images.githubusercontent.com/62376157/128297296-01d87417-8ff3-4f82-8623-dc7bfbe9daaf.png)

and you can extract de data and your labels:

datos=epochs_train.load_data()
id_to_class_dict = {value:key for key, value in datos.event_id.items()}

With all this, I would obtain the matrix which are your inputs and your labels, if you want save it in a dataframe:

datos = epochs_train.load_data()
id_to_class_dict = {value:key for key, value in datos.event_id.items()}

objective_stages = ['Sleep stage N1','Sleep stage N2','Sleep stage N3','Sleep stage R','Sleep stage W']
epochs_to_df = [a for a in datos]
epochs_to_df = []
label_to_df = []
for epoch, label in zip(datos, datos.events[:,-1]):
    # print(label)
    label = id_to_class_dict[label]
    if label in objective_stages:
        # print("True")
        epochs_to_df.append(epoch)
        label_to_df.append(label)
dict_to_df = {"Epoch":epochs_to_df,"Label":label_to_df}
df1=pd.DataFrame(dict_to_df)

But if you use Autoreject library:

import autoreject 

dig_montage = mne.channels.make_standard_montage('standard_1020')
data.drop_channels([ch for ch in data.ch_names
                   if ch not in dig_montage.ch_names])
data.set_montage(dig_montage)  # use the standard montage
epochs = mne.make_fixed_length_epochs(data, duration=30, preload=True)
ar = autoreject.AutoReject(n_interpolate=[1, 2, 3, 4], random_state=11,
                           n_jobs=1, verbose='tqdm')
ar.fit(epochs[:20])

epochs_ar, reject_log = ar.transform(epochs, return_log=True)


If i do epochs_ar.load_data()    I don't get the information with annotations like before with epochs_train.load_data()...:

![image](https://user-images.githubusercontent.com/62376157/128297941-6abe94ae-7bda-417d-9a0a-d50b2ac60b92.png)

**Comment by jasmainak** [comment]
@PARODBE honestly I don't understand what you're trying to do. It's difficult to help if you don't provide a full script and data that I can copy-paste and run on my computer. See: http://sscce.org/

I've edited your comment so it's formatted properly.

Based on what you've said so far, I suspect you observe this difference because you have a different workflow when using Autoreject:

```py
epochs = mne.make_fixed_length_epochs(data, duration=30, preload=True)
```

as opposed to:

```py
events, events_id = mne.events_from_annotations(data)
mne.Epochs(raw, events, ...)
```

Also you should name your variables consistently as in MNE-Python. You're calling `epochs` and `raw` as `data` which is very confusing when reading.

**Comment by hoechenberger** [comment]
> Based on what you've said so far, I suspect you observe this difference because you have a different workflow when using Autoreject:
> 
> ```python
> epochs = mne.make_fixed_length_epochs(data, duration=30, preload=True)
> ```

Yes, I suspect this is the issue here, too.
